src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/caseload/ApproverCaseloadService.kt

```diff
@@ -3,24 +3,141 @@ package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.caseload
+          submittedByFullName = licenceSummary?.submittedByFullName,
+          releaseDate = releaseDate,
+          urgentApproval = licenceSummary?.isDueToBeReleasedInTheNextTwoWorkingDays,
+          approvedBy = licenceSummary?.approvedByName,
+          approvedOn = licenceSummary?.approvedDate,
+          isDueForEarlyRelease = licenceSummary?.isDueForEarlyRelease,
+        ),
+      )
+    }
+
+    val comUsernames = caseload.mapNotNull { it.comUsernameOnLicence }
+    val coms = communityApiClient.getStaffDetailsByUsername(comUsernames)
+
+    val approvalCases = caseload.map {
+      ApprovalCase(
+        licenceId = it.licenceSummary?.licenceId,
+        name = it.licenceSummary?.name,
+        prisonerNumber = it.licenceSummary?.prisonerNumber,
+        submittedByFullName = it.licenceSummary?.submittedByFullName,
+        releaseDate = it.licenceSummary?.releaseDate,
+        urgentApproval = it.licenceSummary?.urgentApproval,
+        approvedBy = it.licenceSummary?.approvedBy,
+        approvedOn = it.licenceSummary?.approvedOn,
+        isDueForEarlyRelease = it.licenceSummary?.isDueForEarlyRelease,
+        probationPractitioner = findProbationPractitioner(it.deliusRecord, it.comUsernameOnLicence, coms),
+      )
+    }.sortedByDescending { it.releaseDate }
+
+    return approvalCases
   }
+}
 
-  companion object {
-    val log: Logger = LoggerFactory.getLogger(this::class.java)
+fun PrisonerSearchPrisoner.getDeliusRecord(deliusRecords: List<OffenderDetail>): OffenderDetail? {
+  return deliusRecords.find { it.otherIds.nomsNumber == this.prisonerNumber }
+}
+
+fun PrisonerSearchPrisoner.getLicenceSummary(licences: List<LicenceSummaryApproverView>): LicenceSummaryApproverView? {
+  val licenceSummaries = licences.filter { it.nomisId == this.prisonerNumber }
+  return if (licenceSummaries.size == 1) licenceSummaries.first() else licenceSummaries.find { it.licenceStatus != LicenceStatus.ACTIVE }
+}
+
+fun PrisonerSearchPrisoner.selectReleaseDate(): LocalDate? {
+  return this.confirmedReleaseDate ?: this.conditionalReleaseDate
+}
+
+private fun findProbationPractitioner(deliusRecord: OffenderDetail?, comUsernameOnLicence: String?, coms: List<User>): ProbationPractitioner? {
+  val responsibleCom = coms.find { com -> com.username?.lowercase() == comUsernameOnLicence?.lowercase() }
+
+  return if (responsibleCom != null) {
+    ProbationPractitioner(
+      staffCode = responsibleCom.staffCode,
+      name = "${responsibleCom.staff?.forenames} ${responsibleCom.staff?.surname}".convertToTitleCase(),
+    )
+  } else {
+    val activeCom = deliusRecord?.offenderManagers?.find { it.active }
+    if (activeCom == null || activeCom.staffDetail.unallocated == true) {
+      return null
+    }
+    ProbationPractitioner(
+      staffCode = activeCom.staffDetail.code,
+      name = "${activeCom.staffDetail.forenames} ${activeCom.staffDetail.surname}".convertToTitleCase(),
+    )
   }
 }
+
+data class ApprovalDetails(
```

Comment:
**suggestion:** could these be private classes? Presumably they are only referenced in this file?